packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -225,7 +225,7 @@ export function getValidationErrors(config: any): string[] {
   if (config !== undefined && typeof config.networks === "object") {
     const hardhatNetwork = config.networks[HARDHAT_NETWORK_NAME];
     if (hardhatNetwork !== undefined) {
-      if (hardhatNetwork.url !== undefined) {
+      if (hardhatNetwork.hasOwnProperty("url")) {
```

Comment:
We usually use `"url" in hardhatNetwork` for this. I know there's a subtle difference between the two things in javascript, but I would do it that way just for consistency.

Reply:
I understand @fvictorio , from what you specify it should be valid to have a `url` property under the `hardhat` network object with value equal to `undefined`. Would it be correct?

Reply:
Hmm, no, it wouldn't. But (AFAIK) the only difference between the idiom I suggested and the one you used is that `hasOwnProperty` doesn't check the prototype chain, but that doesn't really matter in this scenario, that's why I think consistency is more important here.

Reply:
What any other clause could be used instead to check if `url` property is present or not? (Even if it is set to a value or as undefined).

Reply:
I think I'm not explaining myself correctly :sweat_smile: All I'm saying is that I'd rather do

```js
if ("url" in hardhatNetwork) {
```

instead of

```js
if (hardhatNetwork.hasOwnProperty("url")) {
```

They are very similar, and in many cases the latter one is the one you want. Since in this case they are pretty much the same, I'd use the first one since it's the one we use in the rest of the codebase.

Reply:
I understand now, thank you for your efforts on this PR review as I want to be an active contributor of this repository. I would do the changes adding a new commit to this PR and would ensure that new tests continue passing.

In this way, `url` parameter could not be a property under `hardhat` object.

Reply:
My 2cents to avoid future confusion:

In this case we do want to use `in`. If we were to use `hasOwnProperty` to validate the absence of a property, all the consumers of the config should be aware of this, and use the config's own properties instead of just doing `config.value`.